Lib/test/test_sys.py

```diff
@@ -1550,7 +1550,7 @@ class C(object): pass
         def func():
             return sys._getframe()
         x = func()
-        check(x, size('3Pi3c7P2ic??2P'))
+        check(x, size('3Pi3c7P4ic??2P'))
```

Comment:
If you ask me this struct definition was already hopelessly out of date and only passed by sheer luck. With your fix it now passes on Mac/Linux but fails on Windows. I think this is a better format string:
```
        check(x, size('3Pi3c 8Pihhc 2P'))
```
I added spaces (which the struct module ignores in the format) to delineate the PyFrameObject header, the `struct _PyInterpreterFrame`, and two stack slots. (We need two slots because the call pushes the callable followed by a `NULL`.) Similar change below but I'll leave it up to you.

Reply:
But note that if we either reduce the size of `tier2_extra_size` to one byte or get rid of it entirely we'd get back to the original size; the middle portion would become either `8Pihcc` or `8Pihc`.